backend/src/modules/votes/services/create.vote.service.ts

```
@@ -28,55 +43,65 @@ export default class CreateVoteServiceImpl implements CreateVoteService {
 			)
 			.lean()
 			.exec();
+		if (!boardUser) throw Error(UPDATE_FAILED);
+		return boardUser;
 	}
 
-	addVoteToCard(boardId: string, cardId: string, userId: string, cardItemId: string) {
-		this.incrementVoteUser(boardId, userId);
-		return this.boardModel
-			.findOneAndUpdate(
-				{
-					_id: boardId,
-					'columns.cards.items._id': cardItemId
-				},
-				{
-					$push: {
-						'columns.$.cards.$[c].items.$[i].votes': userId
+	async addVoteToCard(boardId: string, cardId: string, userId: string, cardItemId: string) {
+		const voteEnable = await this.voteEnable(boardId, userId);
+		if (voteEnable) {
+			await this.incrementVoteUser(boardId, userId);
+			return this.boardModel
+				.findOneAndUpdate(
+					{
+						_id: boardId,
+						'columns.cards.items._id': cardItemId
 					},
-					$inc: { totalUsedVotes: 1 }
-				},
-				{
-					arrayFilters: [{ 'c._id': cardId }, { 'i._id': cardItemId }],
-					new: true
-				}
-			)
-			.populate({
-				path: 'users',
-				select: 'user role votesCount -board',
-				populate: { path: 'user', select: 'firstName lastName _id' }
-			})
-			.lean()
-			.exec();
+					{
+						$push: {
+							'columns.$.cards.$[c].items.$[i].votes': userId
+						},
+						$inc: { totalUsedVotes: 1 }
+					},
+					{
+						arrayFilters: [{ 'c._id': cardId }, { 'i._id': cardItemId }],
+						new: true
+					}
+				)
+				.populate({
+					path: 'users',
+					select: 'user role votesCount -board',
+					populate: { path: 'user', select: 'firstName lastName _id' }
+				})
+				.lean()
+				.exec();
+		}
+		return null;
```

Instead return `null`, why not return an exception?

```ts
throw new BadRequestException('Error adding a vote')
```

**Note:** maybe you can use generic messages or something else.